src/data/mystery-encounter-requirements.ts

```diff
@@ -98,141 +157,743 @@ export class PartySizeRequirement implements EncounterRequirement {
 
     return true;
   }
+
+  getMatchingDialogueToken(scene:BattleScene): [RegExp, string] {
+    return [new RegExp("@ec\{partySize\\}", "gi"), scene.getParty().length.toString()];
+  }
 }
 
-export class PartySpeciesRequirement implements EncounterRequirement {
-  requiredPartyPokemon: Species[];
+export class PersistentModifierRequirement extends EncounterSceneRequirement {
+  requiredItems?: ModifierType[]; // TODO: not implemented
+  constructor(item: ModifierType | ModifierType[]) {
+    super();
+    if (item instanceof Array) {
+      this.requiredItems = item;
+    } else {
+      this.requiredItems = [];
+      this.requiredItems.push(item);
+    }
+  }
 
-  /**
-   * Party contains at least one of the specified species
-   * To check if party contains multiple different species or species ranges, use multiple instances of this class
-   * @param species
-   */
-  constructor(species: Species | Species[]) {
+  meetsRequirement(scene: BattleScene): boolean {
+    const items = scene.modifiers;
+
+    if (!isNullOrUndefined(items) && this?.requiredItems.length > 0 && this.requiredItems.filter((searchingMod) =>
+      items.filter((itemInScene) => itemInScene.type.id === searchingMod.id).length > 0).length === 0) {
+      return false;
+    }
+    return true;
+  }
+
+  getMatchingDialogueToken(scene:BattleScene): [RegExp, string] {
+    const requiredItemsInInventory = this.requiredItems.filter((a) => {
+      scene.modifiers.filter((itemInScene)  => itemInScene.type.id === a.id).length > 0;
+    });
+    if (requiredItemsInInventory.length > 0) {
+      return [new RegExp("@ec\{requiredItem\\}", "gi"), requiredItemsInInventory[0].name];
+    }
+    return null;
+  }
+
+}
+
+export class MoneyRequirement extends EncounterSceneRequirement {
+  requiredMoney: number;
+
+  constructor(requiredMoney: number) {
+    super();
+    this.requiredMoney = requiredMoney;
+  }
+
+  meetsRequirement(scene: BattleScene): boolean {
+    const money = scene.money;
+    if (!isNullOrUndefined(money) && this?.requiredMoney > 0 && this.requiredMoney > money) {
+      return false;
+    }
+    return true;
+  }
+
+  getMatchingDialogueToken(scene:BattleScene): [RegExp, string] {
+    return [new RegExp("@ec\{money\\}", "gi"), "₽" + scene.money.toString()];
+  }
+}
+
+export class SpeciesRequirement extends EncounterPokemonRequirement {
+  requiredSpecies: Species[];
+  minNumberOfPokemon:number;
+  invertQuery:boolean;
+
+  constructor(species: Species | Species[], minNumberOfPokemon: number = 1, invertQuery: boolean = false) {
+    super();
+    this.minNumberOfPokemon = minNumberOfPokemon;
```

Comment:
Yeah, if we want to specify a party minimum we should do that via the party size requirement. This should just fail meets requirements if party size is 0